policy.go

```diff
@@ -129,45 +133,33 @@ func (p *defaultPolicy[V]) Add(key uint64, cost int64) ([]*Item[V], bool) {
 	// incHits is the hit count for the incoming item.
 	incHits := p.admit.Estimate(key)
 	// sample is the eviction candidate pool to be filled via random sampling.
-	// TODO: perhaps we should use a min heap here. Right now our time
-	// complexity is N for finding the min. Min heap should bring it down to
-	// O(lg N).
-	sample := make([]*policyPair, 0, lfuSample)
+	sample := NewMinHeap[policyPair]()
```

Comment:
Just passing by here 
any reasons to use custom min heap implementation instead of [std](https://pkg.go.dev/container/heap)?

Second point: If the goal of this change is to improve performance, it would be helpful to have benchmark results.